### ISSUE TITLE
Dialyzer Complains

### ISSUE THREAD

**Issue by mattjg908** [RandomPassword/Elixir#1]
Hi, first I'd like to thank you for your work on this project.

I am having an issue where Dialyzer complains when I include `RandomPassword`. I made a simple app [here](https://github.com/mattjg908/effective_meme) to demonstrate the issue. The TLDR is that if you `mix dialyzer`, you will get an error:
```
The function call will not succeed.

CryptoRand.shuffle(_ :: bitstring(), nil)

breaks the contract
(Enumerable.t() | String.t(), (non_neg_integer() -> binary())) :: [any()] | String.t()

________________________________________________________________________________
done (warnings were emitted)
Halting VM with exit status 2
```

I believe the issue is resolved by making something like this change [here](https://github.com/RandomPassword/Elixir/blob/master/lib/random_password.ex#L282) ([CryptoRand.shuffle/2 ](https://github.com/CryptoRand/Elixir/blob/master/lib/crypto_rand.ex#L110)expects a function for the second arg):
```Elixir
       def generate do
         shuffle =
-          &if @random_password_bytes,
+          &if is_function(@random_password_bytes),
             do: CryptoRand.shuffle(&1, @random_password_bytes),
             else: CryptoRand.shuffle(&1)
```

Again, I have not looked deeply into the issue so forgive me if there is some context I'm missing in suggesting this change. Thanks!

**Comment by dingosky** [comment]
Resolved with version 1.1 (commit b623e15).

The version 1.1 upgrade uses `Puid` v2.0 and no longer uses the `CryptoRand` library, which has been removed from `deps`. `Elixir.Enum.shuffle/1` is used to create a permutation of the random strings generated using `Puid`. As long as those random strings are generated using cryptographically strong random bytes (the `Puid` default), a PRNG shuffle performed by `Elixir.Enum.shuffle/1` does not effect the security of the random password's entropy bits.

And thanks for pointing out the library had dialyzer issues. I've ensured both dialyzer and test code coverage are both 100% happy in this latest release.
